src/main/java/seedu/commands/DeleteCommand.java

```diff
@@ -74,12 +76,28 @@ private <T> void deleteEntry(String[] arguments, ArrayList<T> arrList) throws Wi
         String stringIndex = arguments[INDEX_OF_ENTRY_INDEX];
         try {
             int index = Integer.parseInt(stringIndex) - 1;
+            T entryToDelete = arrList.get(index);
             arrList.remove(index);
+            editListTotalAfterDelete(arrList, entryToDelete);
         } catch (NumberFormatException e) {
             throw new WiagiInvalidInputException(INDEX_NOT_INTEGER + DELETE_COMMAND_FORMAT);
         } catch (IndexOutOfBoundsException e) {
             throw new WiagiInvalidIndexException(INDEX_OUT_OF_BOUNDS);
         }
         Ui.printWithTab("Successfully deleted!");
     }
+
+    private <T> void editListTotalAfterDelete(ArrayList<T> arrList, T entryToDelete) {
+        if (arrList instanceof IncomeList) {
+            IncomeList incomes = (IncomeList) arrList;
+            double amountToBeRemoved = ((Income) entryToDelete).getAmount();
+            double updatedAmount = incomes.getTotal() - amountToBeRemoved;
+            incomes.setTotal(updatedAmount);
```

Comment:
just to check is there a need to format this to 2dp

Reply:
Honestly I am not sure, since all the amount involved in the computation should alr be in 2dp, but can round to be safe

Reply:
Resolved in `setTotal` method